osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/config/HibernateNamingStrategy.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2019 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.shared.application.config;
+
+import java.util.Locale;
+
+import org.hibernate.boot.model.naming.Identifier;
+import org.hibernate.boot.model.naming.PhysicalNamingStrategyStandardImpl;
+import org.hibernate.engine.jdbc.env.spi.JdbcEnvironment;
+
+public class HibernateNamingStrategy extends PhysicalNamingStrategyStandardImpl {
+
+    /**
+     * Default serial version UID is required.
+     */
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * Singleton instance of this class.
+     */
+    public static final HibernateNamingStrategy INSTANCE = new HibernateNamingStrategy();
+
+    @Override
+    public Identifier toPhysicalTableName(final Identifier name, final JdbcEnvironment context) {
+        return new Identifier(addUnderscores(name.getText()), name.isQuoted());
+    }
+
+    @Override
+    public Identifier toPhysicalColumnName(final Identifier name, final JdbcEnvironment context) {
+        return new Identifier(addUnderscores(name.getText()), name.isQuoted());
+    }
+
+    /**
+     * Copied from the
+     * {@code org.hibernate.cfg.ImprovedNamingStrategy) class to ensure same behavior.
+     */
+    protected static String addUnderscores(final String name) {
+        final StringBuilder buf = new StringBuilder(name.replace('.', '_'));
+        for (int i = 1; i < buf.length() - 1; i++) {
+            if (Character.isLowerCase(buf.charAt(i - 1)) && Character.isUpperCase(buf.charAt(i))
+                    && Character.isLowerCase(buf.charAt(i + 1))) {
+                buf.insert(i++, '_');
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor the code in order to not assign to this loop counter from within the loop body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AForLoopCounterChangedCheck)

Reply:
This code is ported from the original Hibernate naming strategy we have been using for a long time. Not going to change the implementation.

Reply:
See also https://www.baeldung.com/hibernate-naming-strategy for an alternative implementation using a regex in stead of loop